src/lib/y2network/interface_defaults.rb

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yaml"
+
+module Y2Network
+  module InterfaceDefaults
+
+    # returns default startmode for a new device
+    #
+    # startmode is returned according product, Arch and current device type
+    def new_device_startmode
+      Yast.import "ProductFeatures"
+
+      product_startmode = Yast::ProductFeatures.GetStringFeature(
+        "network",
+        "startmode"
+      )
+
+      startmode = case product_startmode
+      when "ifplugd"
+        if replace_ifplugd?
```

Comment:
Where is this method defined?

Reply:
mostprobably in LanItems. I haven't tested this startmode yet.